sds/src/proximity_keywords.rs

```diff
@@ -317,6 +346,101 @@ fn calculate_keyword_pattern(keyword: &str) -> Ast {
     })
 }
 
+#[derive(Clone, Copy)]
+enum CharType {
+    Regular,
+    Uppercase,
+    Separator,
+}
+
+fn get_chars_type(prev: &char, current: &char, next: &char) -> (CharType, CharType, CharType) {
```

Comment:
`get_chars_type` is a repetition of 3 times the same function for each character. I think it's easier to read if `get_chars_type` is updated to be only applied to a single character. This method will be simplified a lot (to look like only `get_single_char_type`) and `standardize_path_chars` won't be much more complicated:

`standardize_path_chars` can look like:
```rust
...
    for chars in chars.windows(3) {
        let current: &char = &chars[1];
        let prev_char = get_chars_type(&chars[0]);
        let current_char = get_chars_type(&chars[1]);
        let next_char = get_chars_type(&chars[2]);

        match (prev_char, current_char, next_char) {
...
```
It's not pretty but I think it's easier to read

Reply:
Agreed, I did the change in [Improve algorithm and reduce window to 2](https://github.com/DataDog/dd-sensitive-data-scanner/pull/52/commits/44c69740dc08a3c43ce532237f5ed8f402bc78b1)